src/ert/ensemble_evaluator/builder/_ensemble.py

```diff
@@ -34,14 +34,27 @@ def __init__(
         self._snapshot = self._create_snapshot()
         self.status = self._snapshot.status
         self._status_tracker = EnsembleStateTracker(self._snapshot.status)
+        self._id: Optional[str] = None
+
+    @property
+    def id_(self) -> str:
+        if not self._id:
+            raise RuntimeError(f"{self} does not have an ID")
+        return self._id
```

Comment:
Not sure if I like the property name. Although I get we can't use `id` (built-in), I don't think this is very readable, as it looks a bit too much like accessing a private property..

Reply:
I think for the sake of consistency (`.id_` is used for all the sub-ensemble components) we can keep `id_`.